src/ui/dashboard.cpp

```diff
@@ -286,35 +289,21 @@ void Dashboard::load() {
 }
 
 void Dashboard::doLoad(const std::string &desc) {
-    YAML::Node tree = YAML::Load(desc);
+    YAML::Node tree    = YAML::Load(desc);
 
-    auto       path = std::filesystem::path(m_desc->source->path) / m_desc->filename;
+    auto       path    = std::filesystem::path(m_desc->source->path) / m_desc->filename;
 
-#ifdef NDEBUG
-#define ERROR_RETURN(msg) \
-    { \
-        fmt::print("Error parsing YAML {}: {}\n{}\n", path.native(), msg, desc); \
-        abort(); \
-    }
-#else
-#define ERROR_RETURN(msg) \
-    { \
-        fmt::print("Error parsing YAML {}: {}\n{}\n", path.native(), msg, desc); \
-        return; \
-    }
-#endif
-
-    auto sources = tree["sources"];
-    if (!sources || !sources.IsSequence()) ERROR_RETURN("sources entry invalid");
+    auto       sources = tree["sources"];
+    if (!sources || !sources.IsSequence()) throw std::runtime_error("sources entry invalid");
 
     for (const auto &s : sources) {
-        if (!s.IsMap()) ERROR_RETURN("source is no map");
+        if (!s.IsMap()) throw std::runtime_error("source is no map");
```

Comment:
If you use `gr::exception(...)`, you would get implicitly also the `std::source_location` and time trace when/where this exception was thrown.